pysoa/common/serializer/msgpack_serializer.py

```diff
@@ -85,6 +109,14 @@ def default(self, obj):
                 self.EXT_TIME,
                 self.STRUCT_TIME.pack(obj.hour, obj.minute, obj.second, obj.microsecond),
             )
+        elif isinstance(obj, decimal.Decimal):
+            obj_str = six.text_type(obj)[:65535].encode('utf-8')
```

Comment:
shouldn't this raise a `TypeError` if length > 65535?

Reply:
Remember, you're talking about a number with more than 65,535 _digits_. Not a number greater than 65,535. The number would either have to be greater than `10^65534` or have a post-decimal precision of `10^-65534`, or some combination of the two. In other words, this will never happen. However...

The built-in string pack types truncate if the string is longer than the length supported by the leading length byte(s), so this mimics that behavior.

Reply:
I agree it's very unlikely to happen with real data. I don't like silent failures. If it'd be protected by an exception, it would be unlikely to be raised with real data. Not gonna block on that though.